jetty-websocket/websocket-core-client/src/main/java/org/eclipse/jetty/websocket/core/client/WebSocketCoreClient.java

```diff
@@ -52,16 +52,25 @@ public WebSocketCoreClient(WebSocketComponents webSocketComponents)
 
     public WebSocketCoreClient(HttpClient httpClient, WebSocketComponents webSocketComponents)
     {
-        if (httpClient == null)
-            httpClient = Objects.requireNonNull(HttpClientProvider.get());
-        if (httpClient.getExecutor() == null)
-            httpClient.setExecutor(webSocketComponents.getExecutor());
-
-        this.classLoader = Thread.currentThread().getContextClassLoader();
-        this.httpClient = httpClient;
-        this.components = webSocketComponents;
-        addBean(httpClient);
-        addBean(webSocketComponents);
+        client = Objects.requireNonNullElse(httpClient, HttpClientProvider.get());
```

Comment:
There should be tests for `client.isStarted()`, `!client.isStarted()` and `webSocketComponents == null`, `webSocketComponents != null` code paths to make assert that the buffer pool and the executor are shared as expected.

Reply:
@lorban I added tests only for the started/!started cases, as you cannot control WebSocketComponents (it's always null).

Reply:
What if you call this constructor with a null `WebSocketComponents`?

Reply:
@lorban it'll be created.